executor/src/witgen/jit/compiler.rs

```diff
@@ -0,0 +1,804 @@
+        let mut known = vec![0; 1];
+        let params = WitgenFunctionParams {
+            data: MutSlice::from(data.as_mut_slice()),
+            known: known.as_mut_ptr(),
+            row_offset: 0,
+            params: Default::default(),
+            call_machine: no_call_machine,
+        };
+        (f.function)(params);
+        assert_eq!(data[0], GoldilocksField::from(7));
+        assert_eq!(data[1], GoldilocksField::from(9));
+        assert_eq!(known[0], 3);
+    }
+
+    #[test]
+    fn load_twice() {
+        let x = cell("x", 0, 0);
+        let effects1 = vec![Effect::Assignment(
+            x.clone(),
+            GoldilocksField::from(7).into(),
+        )];
+        let effects2 = vec![Effect::Assignment(
+            x.clone(),
+            GoldilocksField::from(9).into(),
+        )];
+        let row_count = 2;
+        let column_count = 2;
+        let data_len = column_count * row_count;
+        let f1 = compile_effects(0, column_count, &[], &effects1).unwrap();
+        let f2 = compile_effects(0, column_count, &[], &effects2).unwrap();
+        let mut data = vec![GoldilocksField::from(0); data_len];
+        let mut known = vec![0; row_count];
+        let params1 = WitgenFunctionParams {
+            data: MutSlice::from(data.as_mut_slice()),
+            known: known.as_mut_ptr(),
+            row_offset: 0,
+            params: Default::default(),
+            call_machine: no_call_machine,
+        };
+        (f1.function)(params1);
+        assert_eq!(data[0], GoldilocksField::from(7));
+        assert_eq!(data[1], GoldilocksField::from(0));
+        assert_eq!(data[2], GoldilocksField::from(0));
+        assert_eq!(data[3], GoldilocksField::from(0));
+        assert_eq!(known, vec![1, 0]);
+        let params2 = WitgenFunctionParams {
+            data: MutSlice::from(data.as_mut_slice()),
+            known: known.as_mut_ptr(),
+            row_offset: 1,
+            params: Default::default(),
+            call_machine: no_call_machine,
+        };
+        (f2.function)(params2);
+        assert_eq!(data[0], GoldilocksField::from(7));
+        assert_eq!(data[1], GoldilocksField::from(0));
+        assert_eq!(data[2], GoldilocksField::from(9));
+        assert_eq!(data[3], GoldilocksField::from(0));
+        assert_eq!(known, vec![1, 1]);
+    }
+
+    #[test]
+    fn field_div_via_integer() {
+        let effects = vec![
+            assignment(&cell("x", 0, 0), number(8).field_div(&number(2))),
+            assignment(&cell("x", 0, 1), number(1 << 60).field_div(&number(8))),
+            assignment(&cell("x", 0, 2), (-number(8)).field_div(&number(2))),
+            assignment(&cell("x", 0, 3), number(8).field_div(&-number(2))),
+            assignment(&cell("x", 0, 4), (-number(8)).field_div(&-number(2))),
+        ];
+        let f = compile_effects(0, 1, &[], &effects).unwrap();
+        let mut data = vec![GoldilocksField::from(0); 5];
+        let mut known = vec![0; 1];
```

Comment:
```suggestion
        let mut known = vec![0; 5];
```
We need at least one `u32` per row. This fixes a segmentation fault when I run on my Mac.